core/storage/concept/generic_map.hpp

```diff
@@ -0,0 +1,28 @@
+/**
+ * Copyright Soramitsu Co., Ltd. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#ifndef KAGOME_GENERIC_MAP_HPP
+#define KAGOME_GENERIC_MAP_HPP
+
+#include "storage/concept/iterable_map.hpp"
+#include "storage/concept/readable_map.hpp"
+#include "storage/concept/writeable_map.hpp"
+
+namespace kagome::storage::concept {
+
+  /**
+   * @brief An abstraction over readable, writeable key-value map.
+   * @tparam K key type
+   * @tparam V value type
+   */
+  template <typename K, typename V>
+  struct GenericMap : public IterableMap<K, V>,
+                      public ReadableMap<K, V>,
+                      public WriteableMap<K, V> {
+    ~GenericMap() override = default;
```

Comment:
If I am correct, there's no need in overriding a destructor here, only in the implementations